code/game/objects/machinery/Sleeper.dm

```diff
@@ -439,6 +439,15 @@
 	else
 		to_chat(user, span_notice("There is no one inside!"))
 
+/obj/machinery/sleeper/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(!occupant)
+		return
+	if(X.status_flags & INCORPOREAL || X.do_actions)
+		return
+	if(!do_after(X, 2 SECONDS))
```

Comment:
can we combine all of these?

Reply:
Do you also think we can have feedback when opening the sleepers?